Makefile

```diff
@@ -203,42 +190,62 @@ $(OUT)/treasuryDAO_storage.tz: src/**
             } \
           ; config_data = \
-            { max_quorum = { numerator = ($(max_quorum) * quorum_denominator)/100n } \
-            ; min_quorum = { numerator = ($(min_quorum) * quorum_denominator)/100n } \
-            ; period = { length = $(period) } \
-            ; proposal_flush_time = $(proposal_flush_time) \
-            ; proposal_expired_time = $(proposal_expired_time) \
-            ; fixed_proposal_fee_in_token = $(fixed_proposal_fee_in_token) \
-            ; quorum_threshold = { numerator = ($(quorum_threshold) * quorum_denominator)/100n } \
-            ; quorum_change = { numerator = ($(quorum_change) * quorum_denominator)/100n } \
-            ; max_quorum_change = { numerator = ($(max_quorum_change) * quorum_denominator)/100n } \
-            ; governance_total_supply = $(governance_total_supply) \
+            { max_quorum = { numerator = (($(call require_defined,max_quorum) : nat) * quorum_denominator)/100n } \
+            ; min_quorum = { numerator = (($(call require_defined,min_quorum) : nat) * quorum_denominator)/100n } \
+            ; period = { length = ($(call require_defined,period) : nat) } \
+            ; proposal_flush_time = ($(call require_defined,proposal_flush_time)  : nat) \
+            ; proposal_expired_time = ($(call require_defined,proposal_expired_time) : nat) \
+            ; fixed_proposal_fee_in_token = ($(call require_defined,fixed_proposal_fee_in_token) : nat) \
+            ; quorum_threshold = { numerator = (($(call require_defined,quorum_threshold) : nat) * quorum_denominator)/100n } \
+            ; quorum_change = { numerator = (($(call require_defined,quorum_change) : nat) * quorum_denominator)/100n } \
+            ; max_quorum_change = { numerator = (($(call require_defined,max_quorum_change) : nat) * quorum_denominator)/100n } \
+            ; governance_total_supply = ($(call require_defined,governance_total_supply) : nat) \
             }} \
-          ; frozen_scale_value = $(frozen_scale_value) \
-          ; frozen_extra_value = $(frozen_extra_value) \
-          ; max_proposal_size = $(max_proposal_size) \
-          ; slash_scale_value = $(slash_scale_value) \
-          ; slash_division_value = $(slash_division_value) \
-          ; min_xtz_amount = $(min_xtz_amount) \
-          ; max_xtz_amount = $(max_xtz_amount) \
+          ; frozen_scale_value = ($(call require_defined,frozen_scale_value) : nat) \
+          ; frozen_extra_value = ($(call require_defined,frozen_extra_value) : nat) \
+          ; max_proposal_size = ($(call require_defined,max_proposal_size) : nat) \
+          ; slash_scale_value = ($(call require_defined,slash_scale_value) : nat) \
+          ; slash_division_value = ($(call require_defined,slash_division_value) : nat) \
+          ; min_xtz_amount = ($(call require_defined,min_xtz_amount) : tez) \
+          ; max_xtz_amount = ($(call require_defined,max_xtz_amount) : tez) \
```

Comment:
These changes to use `require_defined` are now unnecessary.

Since there is a default value there will never be a case in which there are undefined.

I mean, it could only happen if one purposefully calls e.g. `make out/treasuryDAO_storage.tz max_xtz_amount=""`, but LIGO will throw a better error in that case anyway.

Note: this comment applies to the other non-necessary changes like this, including the other storage targets.

Reply:
I kept it in case when we need it in the future. But I'm good with removing it for now. I pushed the update and handled other changes as well.